### PR TITLE
refactor!: nuxt 3 only support

### PR DESCRIPTION
**Pull request by kylegl**
In this PR,  I have rewritten the Nuxt 3 component using the Composition API. To ensure type safety, I had to remove the Nuxt 2 module. I propose that we handle this similarly to [Nuxt Image](https://github.com/nuxt/image/blob/main/README.md) by maintaining two separate branches: `v0`  for Nuxt 2 and `main` for Nuxt 3.

There are few notable changes:

* The module name in the Nuxt config has been changed from `google-adsense` to `googleAdsense`. This adjustment was made to resolve type errors caused by the hyphen.
* The module config now accepts a new option called `hideUnfilled`. This option will hide any unfilled ads, as recommended in the Google AdSense documentation.
* The `adFormat` component prop no longer defaults to `auto`. I encountered scenarios while using the module where I needed adFormat to be undefined, and this caused issues. A default global `adFormat` can be set in the module config options and overridden with component props  if needed.
* The component now exposes the `showAd` and  `updateAd` methods, as well as an `isUnfilled` computed ref for user customization if needed.

This PR fixes issue #174  #173

### CODE REVIEW

**Comment by kylegl** [comment]
@manniL

**Comment by manniL** [comment]
@kylegl Thanks for the PR! This means that the module is not Nuxt 2 compatible anymore, right? Hence we'd need a new major version.

File: README.md [hunk 1]
```diff
@@ -130,14 +152,24 @@ to handle updating ads on progressive web applications:
 
 Each time a new advertisement is requested, the AdSense parameter `data-ad-region` is
 updated to a random value. For this reason, you cannot set the `data-ad-region` attribute
-on the `<adsbygoogle />` component.
-
-For test mode, the following blog was used as a reference:
```
Comment (manniL) [inline README.md:135]: This is still applied in the component, right?

File: package.json [hunk 2]
```diff
@@ -1,12 +1,13 @@
 {
   "name": "@nuxtjs/google-adsense",
-  "version": "2.1.0",
+  "version": "1.0.0",
```
Comment (manniL) [inline package.json:3]: Probably has to go to `3.x` then

File: README.md (hunk 1 above)
Comment (kylegl) [reply README.md:135]: Yes it is still applied. I just Capitalized the component name and removed the link to the blog that 404s here.

**Comment by kylegl** [comment]
> @kylegl Thanks for the PR! Good work 👏
> 
> This means that the module is not Nuxt 2 compatible anymore, right? Hence we'd need a new major version (as mentioned in the updated README).

Correct, this pr removes Nuxt 2 compatibility. I wanted to get your thoughts on how to handle this.

Did you happen to see how the Nuxt Image module handled dropping Nuxt 2? They made a branch for Nuxt 2 and branch for Nuxt 3.  Then to install Nuxt 2 module is  `pnpm install  @nuxt/image` and to install Nuxt 3 is `@nuxt/
image@rc`. 

I think having separate branches might be better because the docs & config are different, but I'm not sure the best way to handle it.

**Comment by manniL** [comment]
> Correct, this pr removes Nuxt 2 compatibility. I wanted to get your thoughts on how to handle this.

> I think having separate branches might be better because the docs & config are different, but I'm not sure the best way to handle it. @manniL I'd be happy to set this up. lmk if I can help in anyway.

I think the easiest is mentioning that the Nuxt 2 version can be found under branch XYZ and the package only supports Nuxt 3 from major version ABC on. 

That shouldn't be an issue at all 👍

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,54 @@
...
+- Update nuxt module builder packages ([a635a93](https://github.com/kylegl/google-adsense/commit/a635a93))
+
+### 🎨 Styles
+
+- Playground ([460ca89](https://github.com/kylegl/google-adsense/commit/460ca89))
+
+### ❤️  Contributors
+
+- Kyle <lauber.kyle@gmail.com>
+
```
Comment (manniL) [inline CHANGELOG.md:52]:
```suggestion
```

Will be auto-generated 👍

File: README.md [hunk 4]
```diff
@@ -1,3 +1,15 @@
+[![npm version][npm-version-src]][npm-version-href]
+[![npm downloads][npm-downloads-src]][npm-downloads-href]
+[![License][license-src]][license-href]
+[![Nuxt][nuxt-src]][nuxt-href]
+
+---
+
+**Note:** This branch is for **Nuxt 3** compatible module. Checkout [`v0` branch](https://github.com/nuxt/image/tree/v0) for **Nuxt 2** support.
```
Comment (manniL) [inline README.md:8]:
```suggestion
**Note:** This branch is for **Nuxt 3** compatible module. Checkout [`legacy-v2` branch](https://github.com/nuxt-modules/google-adsense/tree/legacy-v2) for **Nuxt 2** support.
```

File: README.md [hunk 5]
```diff
@@ -24,7 +36,7 @@ Using top level options:
     ['@nuxtjs/google-adsense']
   ],
 
- 'google-adsense': {
+ 'googleAdsense': {
```
Comment (manniL) [inline README.md:39]:
```suggestion
 googleAdsense: {
```

Now the `''` are not needed anymore ☺️

File: README.md [hunk 6]
```diff
@@ -38,13 +50,13 @@ Using runtime config:
     ['@nuxtjs/google-adsense']
   ],
 
-  'google-adsense': {
+  'googleAdsense': {
```
Comment (manniL) [inline README.md:53]:
```suggestion
  googleAdsense: {
```

File: README.md [hunk 7]
```diff
@@ -38,13 +50,13 @@ Using runtime config:
...
     onPageLoad: false,
     pageLevelAds: false,
   },
 
   publicRuntimeConfig: {
-    'google-adsense': {
+    'googleAdsense': {
```
Comment (manniL) [inline README.md:59]:
```suggestion
    googleAdsense: {
```

File: README.md [hunk 8]
```diff
@@ -130,14 +152,24 @@ to handle updating ads on progressive web applications:
...
-For test mode, the following blog was used as a reference:
-- https://www.thedev.blog/3087/test-adsense-ads-safely-without-violating-adsense-tos/
+on the `<Adsbygoogle />` component.
 
 
 ## License
 
 [MIT License](./LICENSE)
 
 Copyright (c) 2017 Troy Morehouse - Nuxt Community
```
Comment (manniL) [inline README.md:162]:
```suggestion
```

Should be enough to link the MIT license above 👍

**Comment by manniL** [comment]
@kylegl created a legacy-v2 branch and annotated some more things! Please tag for a re-review, then I think it is good to merge 👍

**Comment by kylegl** [comment]
Excellent, everything looks good to go!

File: package.json [hunk 9]
```diff
@@ -1,12 +1,13 @@
 {
-  "name": "@nuxtjs/google-adsense",
-  "version": "2.1.0",
+  "name": "@kylegl/google-adsense",
```
Comment (manniL) [inline package.json:2]:
```suggestion
  "name": "@nuxtjs/google-adsense",
```

File: package.json [hunk 10]
```diff
@@ -1,12 +1,13 @@
...
+  "version": "0.0.1",
```
Comment (manniL) [inline package.json:3]:
```suggestion
  "version": "2.1.0",
```

will be bumped automatically with changelog gen in a bit

File: package.json [hunk 11]
```diff
@@ -1,12 +1,13 @@
...
   "description": "Google AdSense Module for Nuxt.js",
+  "repository": "https://github.com/kylegl/google-adsense.git",
```
Comment (manniL) [inline package.json:5]:
```suggestion
  "repository": "https://github.com/nuxt-modules/google-adsense.git",
```

**Pull request merged**